### ISSUE TITLE
Cannot import packages added using sys.modules.update

### ISSUE THREAD

**Issue by artificial-aidan** [microsoft/pyright#3097]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
GRPC imports an aio package using the following code in `grpc/__init__.py`:

```
# Prevents import order issue in the case of renamed path.
if sys.version_info >= (3, 6) and __name__ == "grpc":
    from grpc import aio  # pylint: disable=ungrouped-imports
    sys.modules.update({'grpc.aio': aio})
```

This makes pyright fail to find the package `grpc.aio`

**To Reproduce**

Create a python file like so:

```
import grpc.aio
```

**Expected behavior**

Pyright should find the import

**Comment by artificial-aidan** [comment]
Not sure if this is related, but they also import packages like this:

```
try:
    import grpc_reflection
    sys.modules.update({'grpc.reflection': grpc_reflection})
except ImportError:
    pass
```

And that doesn't work either.

**Comment by erictraut** [comment]
Which package is this? I tried `pip install grpc`, but I don't think that's the package you're talking about here.

**Comment by artificial-aidan** [comment]
Sorry, `grpcio`

**Comment by erictraut** [comment]
It looks like `grpcio` is not a typed library. Do you have type stubs installed for it?

Are you using all default settings for pyright, or do you have a pyrightconfig.json file that overrides some of the defaults? In particular, do you have `useLibraryCodeForTypes` enabled?

When you say "cannot import packages", what behavior are you seeing? Are you seeing the import interpreted as an `Unknown` type? Are you receiving an error message? If so, which one?

I tried to repro the problem as reported, and I'm not seeing anything unexpected.

**Comment by artificial-aidan** [comment]
I do not have stubs for it installed. Looks like there is some additional work going on to generate stubs for the aio aspect of it. While I'd love to have full type stubs, even just not having to ignore it for now would be great.

I am using it through Pylance and standalone. Get the error with both.

I tried with and without the `useLibraryCodeForTypes` enabled.

I'm getting the error message: `error: Import "grpc.aio" could not be resolved `

**Comment by artificial-aidan** [comment]
Hmm, I also just realized in a different project I don't have this issue. Let me see if I can sort out why i'm only seeing it in one project.

**Comment by artificial-aidan** [comment]
Ok now I'm stumped. I can't reproduce this in a standalone project either. Only in this virtual env. I don't believe this to be a pyright bug, but I will report back here for the next time someone runs into this.

**Comment by artificial-aidan** [comment]
Any debug output I should look at? `grpc` imports fine, but `grpc.aio` does not.

**Comment by erictraut** [comment]
Yeah, I'm pretty sure this isn't a bug in pyright.

You could enable `verboseOutput` in the [pyrightconfig.json](https://github.com/microsoft/pyright/blob/main/docs/configuration.md) file. This will emit additional diagnostics related to import resolution.

**Comment by artificial-aidan** [comment]
Ok. So I finally sorted out the problem.

I had [grpc-stubs](https://github.com/shabbyrobe/grpc-stubs) installed, which are not compatible with `grpc.aio`. 

If I remove the `grpc-stubs` library, then I can resolve the import again.

Is there a way to annotate that a package should only provide stubs for part of another package, or some other way to maintain the default type inference.

Basically, pylance gives me some amount of type info when I don't have the stubs installed, but once the stubs are installed it can't find `grpc.aio` module at all.

**Comment by erictraut** [comment]
If you believe this is a bug in the stub package, you could contact the maintainer of that package and report the bug.
